docs/learn/expert-guides/key-manager/get-controller-permissions.md

```diff
@@ -5,185 +5,169 @@ sidebar_position: 1
+
+:::
 
 ## Setup
 
+First, we will install the 🛠️[`erc725.js`](../../../tools/erc725js/getting-started.md) library to easily query and decode the 🗂️[ERC725Y data storage](../../../standards/lsp-background/erc725.md#erc725y-generic-data-keyvalue-store) of the Universal Profile. You can then initialize a Universal Profile and connect to the LUKSO L16 network.
+
 ```bash
 npm install @erc725/erc725.js
 ```
 
-## Step 1 - setup web3.js and erc725.js
-
-The first step is to set up both and _web3.js_ and _erc725.js_, and connect to the LUKSO L16 network. We will also need the address of the Universal Profile that we want to get the controller addresses from.
-
 ```js
 import { ERC725 } from '@erc725/erc725.js';
 import LSP6Schema from '@erc725/erc725.js/schemas/LSP6KeyManager.json';
-import Web3 from 'web3';
-
-// setup
-const myUniversalProfileAddress = '0xC26508178c4D7d3Ad43Dcb9F9bb1fab9ceeD58B5';
-const RPC_ENDPOINT = 'https://rpc.testnet.lukso.network';
 
-// step 1 - setup erc725.js
-const web3 = new Web3(RPC_ENDPOINT);
+// Set up Universal Profile on the LUKSO Testnet
 const erc725 = new ERC725(
   LSP6Schema,
-  myUniversalProfileAddress,
-  web3.currentProvider,
+  // Sample Profile Address
+  '0xEda145b45f76EDB44F112B0d46654044E7B8F319',
+  // LUKSO Testnet RPC
+  'https://rpc.testnet.lukso.network',
 );
 ```
 
-## Step 2 - Get the list of controller addresses
+## Get the Controller Addresses
 
-The next step is to retrieve the list of controller addresses. In other words, addresses with some permissions can interact with the Universal Profile.
+After setting up the Universal Profile, you can retrieve the list of controller addressess by querying the [`AddressPermissions[]`](../../../standards/universal-profile/lsp6-key-manager.md#retrieving-addresses-with-permissions) key of the 🗂️[ERC725Y data storage](../../../standards/lsp-background/erc725.md#erc725y-generic-data-keyvalue-store). This will return an array with all controller addresses of the Universal Profiles:
 
-All the controller addresses are listed under the data key `AddressPermissions[]`. We can retrieve them by:
+```js
+async function getPermissionedAddresses() {
+  // Get the list of addresses that have permissions on the Universal Profile
+  const result = await erc725.getData('AddressPermissions[]');
 
-1. querying the `AddressPermissions[]` to get the array length and know **how many controller addresses** have some permissions on the UP.
-2. querying each index of the `AddressPermissions[index]` array to retrieve each address individually.
+  if (!result) {
+    console.error('No controllers listed under this Universal Profile ');
+  }
+}
+```
 
-Thanks to _erc725.js_, we can do that very easily. The library will do both steps for us and will returns the full list of controller addresses.
+<Details>
+    <summary>Show <code>AddressPermissions[]</code> value</summary>
```

Comment:
I would keep this close to the code, it is the value of "result" (which btw could be renames to "permissions"

<img width="671" alt="image" src="https://github.com/lukso-network/docs/assets/477945/bbb208c4-0418-4ad0-b298-7a09bb2c09f1">

The output is quite small so im not sure we even need a Details box.

Btw, please use `<details>` as the issue with this is gone (revert to previous version of react mdx)

Reply:
im not sure these emojis 🗂️ are really needed here, same for 🛠️